### ISSUE TITLE
Mesh_3 computes 1000 initial points just to have an estimation of a bounding box

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#279]
In `Mesh_3/include/CGAL/Mesh_3/Mesher_3.h`, we can find:
```
  //=====================================
  // Bounding box estimation
  //=====================================
  typedef std::vector<std::pair<Point, Index> > Points_vector;
  Points_vector random_points_on_surface;
  r_oracle_.construct_initial_points_object()(
    std::back_inserter(random_points_on_surface), 1000);
  typename Points_vector::const_iterator
    it = random_points_on_surface.begin(),
    it_end = random_points_on_surface.end();
  Bbox_3 estimated_bbox = it->first.bbox();
  ++it;
  for( ; it != it_end ; ++it)
    estimated_bbox = estimated_bbox + it->first.bbox();

  Base::set_bbox(estimated_bbox);
```

In case the parallelism is not used, `Base::set_bbox` does nothing, and the `estimated_bbox` was just computed for nothing.

**Comment by lrineau** [comment]
My affirmation was wrong: `estimated_bbox` is also used to compute the far points.

We should work with @cjamin to find another way: the bounding box could be part of the [`MeshDomain_3`](http://doc.cgal.org/4.7/Mesh_3/classMeshDomain__3.html) concept.

**Comment by lrineau** [comment]
See 8035aa83ac5c673057eb43191c456b780f57c5dc for a first simple fix.

**Comment by cjamin** [comment]
Your fix is good enough for now, but I agree this estimation is not perfect. I did it this way because I only needed a rough estimation of the bounding box.

**Comment by lrineau** [comment]
Fixed by PR #559.
